packages/edit-site/src/components/secondary-sidebar/list-view-sidebar.js

```diff
@@ -59,7 +62,7 @@ export default function ListViewSidebar() {
 					focusOnMountRef,
 				] ) }
 			>
-				<ListView />
+				<ListView MoreMenuComponent={ BlockSettingsDropdown } />
```

Comment:
I'm not sure why, but passing a component as a prop seems like an anti-pattern. I would expect something more like this.

```suggestion
				<ListView MoreMenuComponent={ (props) => (<BlockSettingsDropdown {...props} />) } />
```

Reply:
It also feels a little weird to include the option at the top-level of the list view. Composition is usually a better way to extend components rather than just adding more options.

Reply:
I'm not sure how this could be done with composition, without still modifying the props of the block. @talldan did you have any thoughts?

Reply:
If there are ways to make it more composable, that would be great. @ajlende maybe you could share some pseudocode of what you're thinking.

The balancing act is deciding how high or low level it should be. There are some more complex internal parts to ListView (the branching structure and the tree grid semantics) that shouldn't be exposed to implementors.

Initially I started thinking about something like this:
```js
<ListView>
    <ListViewRows>
        { ( block ) => (
            <ListViewRow>
                <ListViewBlockDetails block={ block } />
                <ListViewBlockSettingsDropdown block={ block } />
            </ListViewRow
        ) }
    </ListViewRows>
</ListView>
```

But I don't know how useful `ListViewRows`, `ListViewRow` and `ListViewBlockDetails` would be (they would always be there), which is when I came to the idea of just being able to add extra controls/columns.